lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealActivity.kt

```diff
@@ -42,7 +71,99 @@ class FindMealActivity : AppCompatActivity() {
             swipeRefreshLayout.isRefreshing = false
             displayMeals(true)
         }
-        displayMeals(false)
+
+        checkLocationPermission()
+    }
+
+    private fun checkLocationPermission() {
+        if (ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) !== PackageManager.PERMISSION_GRANTED
+        ) {
+            ActivityCompat.requestPermissions(
+                this,
+                arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
+                LOCATION_PERMISSION_REQUEST_CODE
+            )
+        } else {
+            updateLastLocation()
+        }
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        if (permissionResultParser.isFineLocationPermissionsGranted(permissions, grantResults)) {
+            updateLastLocation()
+            startLocationUpdates()
+        } else {
+            // Ask user to update settings
+        }
+    }
+
+    private fun updateLastLocation() {
+        fusedLocationProviderClient.lastLocation.addOnSuccessListener { location: Location ->
+            lastLocation = location
+            displayMeals(false)
+        }
+    }
+
+    private fun startLocationUpdates() {
+        val locationRequest = LocationRequest()
+
+        locationRequest.interval = 100000
+        locationRequest.fastestInterval = 5000
+        locationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+
+        if (!hasSetupLocationUpdates) {
+            setupLocationUpdates(locationRequest)
+        }
+        fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback, null)
+    }
+
+    private fun setupLocationUpdates(locationRequest: LocationRequest) {
+        val builder = LocationSettingsRequest.Builder()
+            .addLocationRequest(locationRequest)
+
+        val client = LocationServices.getSettingsClient(this)
+        val task = client.checkLocationSettings(builder.build())
```

Comment:
Hmm is this really setting up location updates, or is it just checking permissions? Since we're only requesting permissions on a successful permissions request, is it still needed?